File: app/src/main/java/com/chooloo/www/callmanager/ui/recents/RecentsPresenter.java
package com.chooloo.www.callmanager.ui.recents;

import com.chooloo.www.callmanager.entity.RecentCall;
import com.chooloo.www.callmanager.ui.cursor.CursorPresenter;

public class RecentsPresenter<V extends RecentsMvpView> extends CursorPresenter<V> implements RecentsMvpPresenter<V> {

    @Override
    public void onRecentItemClick(RecentCall recentCall) {
    }

    @Override
    public boolean onRecentItemLongClick(RecentCall recentCall) {
        return true;
        // TODO implement
    }
}

File: app/src/main/java/com/chooloo/www/callmanager/ui/recents/RecentsPresenter.kt
package com.chooloo.www.callmanager.ui.recents

import com.chooloo.www.callmanager.entity.Recent
import com.chooloo.www.callmanager.ui.list.ListPresenter

class RecentsPresenter<V : RecentsMvpView> : ListPresenter<V>(), RecentsMvpPresenter<V> {
    override fun onRecentItemClick(recent: Recent) {}
    override fun onRecentItemLongClick(recent: Recent) {}
}